haskell/cabal.bzl

```diff
@@ -244,7 +227,6 @@ def _haskell_cabal_library_impl(ctx):
         env = c.env,
```

Comment:
@regnat Re the CI failure. `c.env` [contains `PATH`](https://github.com/tweag/rules_haskell/pull/1097/files#diff-b8d0916c99703c813c8bcf9b8585a5cfR97). Maybe this overrides the default `PATH` which would have contained the python interpreter. The [log output](https://buildkite.com/tweag-1/rules-haskell/builds/363#3b434151-99db-4375-bd11-eb11bb2034ac/16-195) suggests that:
```
exec env - \
--
  | LANG=en_US.UTF-8 \
  | LD_LIBRARY_PATH='' \
  | LIBRARY_PATH='' \
  | PATH='' \
  | bazel-out/host/bin/external/rules_haskell/haskell/cabal_wrapper
```

Reply:
Aaaah, that might be it indeed. But that would be a bit strange since `env -` is supposed to clean the environment anyways :thinking:

But a quick test shows that `env -` and `env - PATH=''` are behaving slightly differently:

```console
$ env - sh -c 'echo $PATH'
/no-such-path
$ env - PATH='' sh -c 'echo $PATH'
env: ‘sh’: No such file or directory
```

Reply:
So you were right indeed, removing `PATH=''` fixes this issue. But now `runghc` fails because it's a shell script expecting some coreutils to be present.
For some magic, `ghc` has a non-empty `PATH` when we run it through `haskell_library` but not here… I'm a bit lost here. That's probably because some system-level init file sets it when we start bash, but I don't see why it dosen't apply to `haskell_cabal_library`

Reply:
The original cabal wrapper [preserved the incoming `PATH`](https://github.com/tweag/rules_haskell/pull/1097/files#diff-b8d0916c99703c813c8bcf9b8585a5cfR89). Have you tried the same here? Meaning, not passing `env` to `run`, but instead setting it in the wrapper itself with a substitution.

Reply:
After a seance of nasty debugging with @Profpatsch we found the root cause:

Shells (at least bash and dash) have this wonderful behavior that if they are started with `PATH` unset they provide a default hardcoded value for it. So using a shell wrapper script works fine because if magically gives us a PATH with all we need inside. But that's not the case for python, hence the difference.

The ~solution~ ugly workaround we found for this was to set `os.environ["PATH"] = subprocess.check_output(["/bin/sh", "-c", "echo $PATH"], env={})` to get the default shell from sh and get back the old behaviour (which is awful but is the way things used to work)